hack/versions/pkg/schema/comment_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+const configFileTemplate = `/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta12
+
+import (
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/util"
+)
+
+%sconst Version string = "skaffold/v1beta12"
+
+// NewSkaffoldConfig creates a SkaffoldConfig
+func NewSkaffoldConfig() util.VersionedConfig {
+	return new(SkaffoldConfig)
+}
+`
+
+var configWithNoComment = fmt.Sprintf(configFileTemplate, "")
+var configWithReleasedComment = fmt.Sprintf(configFileTemplate, releasedComment+"\n")
+var configWithUnreleasedComment = fmt.Sprintf(configFileTemplate, unreleasedComment+"\n")
+
+func TestAddComment(t *testing.T) {
+
+	tcs := []struct {
+		name     string
+		orig     string
+		expected string
+		released bool
+	}{
+		{
+			name:     "unreleased comment added on file",
```

Comment:
It would be easy to also cover `historicalVersionComments`  here by overriding it. 
The test can serve as a good documentation on how to use it.

Reply:
As I'm thinking about this - I should rename this to `knownComments` and we should fail on a file if it doesn't have a `knownComment` as the first comment.